runtime/common/src/main/java/org/apache/nemo/runtime/common/metric/TaskMetric.java

```diff
@@ -34,6 +36,8 @@
   private String containerId = "";
   private int scheduleAttempt = -1;
   private List<StateTransitionEvent<TaskState.State>> stateTransitionEvents = new ArrayList<>();
+  private final Map<String, List<StreamMetric>> streamMetrics = new HashMap<>();
```

Comment:
What is the key of this map?

Reply:
source vertex id of data fetcher. It means where is the data come from. 

I tried to use taskId instead. But there are no way to get previous task id in the task.